### PR TITLE
Tau and updated jx cases

### PR DESCRIPTION
**Pull request by lucas-spangher-1**
Hey man, 

I've done the following:

- updated the Jinxiang case function and tested it that it produces sensible output that cooperates with your function
---- a note, there's no seed in it so it randomizes output, not sure if you want to rectify that or not. 
- implemented the tau windowing. Right now tau values are set, but with a small modification they could be passed in and optimized. 
- in my opinion I've tidied up the ModelReadyDataset while not changing any functionality by: 
- converting any Timedelta columns to seconds (maybe it was just my local dataset that had it, not sure.) 
- replacing your rand = random.Random() with just random.random(), a bit simpler as that just samples. 
- replacing len with length to avoid renaming the function, etc. 

There wasn't much more I felt license to do without messing things up, I apologize. I thought about overhauling it to do predictive state pretraining but don't have a good enough handle yet on how normal training happens.

### CODE REVIEW

**Comment by Aphoh** [comment]
random.Random was specifically used so that it has reproducible output. Otherwise, reconstructing the specific dataset is hard when random is used elsewhere. Previously it had a seed passed in and seeded the random.Random object with it.

File: datamodules/lucas.py [hunk 1]
```diff
@@ -13,13 +13,14 @@
 import os
 
 
+# TODO: check output of collate fn is comparable to the output of my collate fn
```
Comment (Aphoh) [inline datamodules/lucas.py:15]: It probably shouldn't be, unless you put the sequence dimension last like this does.

File: datamodules/lucas.py [hunk 2]
```diff
@@ -36,7 +37,7 @@ class LucasDataModule(pl.LightningDataModule):
     DataModule for Lucas' fusion dataset.
     """
 
-    DATA_FILENAME = "lucas_data_f32.pickle"
+    DATA_FILENAME = "/Users/cusg/Documents/MIT PSFC/projects/Disruption Prediction/hdl_improvement/Model_training/Full_HDL_dataset_unnormalized_no_nan_column_names_w_shot_and_time.pickle"
```
Comment (Aphoh) [inline datamodules/lucas.py]: Did you make modifications to the dataset? If so I can put them in cloudflare and set up the downloading in the LucasDataLoader

File: datamodules/lucas.py [hunk 3]
```diff
@@ -55,9 +56,10 @@ def __init__(
         val_percent: float = 0.1,
         num_workers: int = 1,
         augment: bool = False,
-        debug: bool = False,
+        debug: bool = True,
```
Comment (Aphoh) [inline datamodules/lucas.py]: Change back to False default

File: datamodules/lucas.py [hunk 4]
```diff
@@ -55,9 +56,10 @@ def __init__(
...
         seed: int = 42,
         len_aug_args: dict = {},
+        taus: dict = {},
```
Comment (Aphoh) [inline datamodules/lucas.py]: This should just be part of the len aug args right?

File: datamodules/lucas.py [hunk 5]
```diff
@@ -134,14 +139,13 @@ def setup(self, stage=None):
...
             new_machine=self.new_machine,
-            seed=self.seed,
         )
 
         if self.debug:
             train_inds = train_inds[:80]
             test_inds = test_inds[:20]
 
-        n_val = int(round(len(train_inds) * self.val_percent))
+        n_val = int(round(len(train_inds) * self.val_percent)) ## maybe cut these off the test set instead?
```
Comment (Aphoh) [inline datamodules/lucas.py]: Nope, val always comes from the train set. Test set is basically the data that doesn't exist yet.

File: datamodules/lucas.py [hunk 6]
```diff
@@ -134,14 +139,13 @@ def setup(self, stage=None):
             dataset=data,
             case_number=self.case_number,
             new_machine=self.new_machine,
-            seed=self.seed,
```
Comment (Aphoh) [inline datamodules/lucas.py:138]: Put this back so it can be reproducible 😁

File: datamodules/lucas_processing.py [hunk 7]
```diff
@@ -60,10 +65,18 @@ def __init__(
                     "Must provide either end_cutoff or end_cutoff_timesteps"
                 )
 
-            d = torch.tensor(shot_df[:shot_end])
+            timedelta_columns = [col for col in shot_df.columns if isinstance(shot_df[col].iloc[0], pd.Timedelta)]
```
Comment (Aphoh) [inline datamodules/lucas_processing.py]: Did you observe this in the version of the dataset that gets downloaded?

File: datamodules/lucas_processing.py [hunk 8]
```diff
@@ -172,98 +189,94 @@ def collate_fn_seq_to_seq(dataset):
     return output
 
 
-def get_train_test_indices_from_Jinxiang_cases(
-    dataset, case_number, new_machine, seed, case8_percentage=0.125
```
Comment (Aphoh) [inline datamodules/lucas_processing.py]: Def put back the seed

File: datamodules/lucas_processing.py [hunk 9]
```diff
@@ -299,21 +312,23 @@ def get_class_weights(train_dataset):
 def length_augmentation(
     x,
     y,
-    len,
-    rand: random.Random,
+    length,
```
Comment (Aphoh) [inline datamodules/lucas_processing.py]: Definitely put the seed back here too!

File: sbatch_submit.sh [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+conda activate will-env
+
+
+cd /home/spangher/ccnn
+. ./venv/bin/activate
+
+# Ignore the bogus local user packages on the host
+export PYTHONUSERBASE=intentionally-disabled
+# Run multiple instantiations of the WandB agent in parallel
+export WANDB_API_KEY=d38dd9743cf22de213611707a16c0ef5beb0ee83
```
Comment (Aphoh) [inline sbatch_submit.sh:23]: Ooops

**Review by Aphoh — CHANGES_REQUESTED** [review]

File: datamodules/lucas.py (hunk 1 above)
Comment (lucas-spangher-1) [reply datamodules/lucas.py:15]: Gotcha -- mine is B x T x C, so yeah, they'd be different. Thanks!

File: datamodules/lucas.py (hunk 2 above)
Comment (lucas-spangher-1) [reply datamodules/lucas.py]: oooo no I don't think I changed any numbers specifically so we're OK

File: datamodules/lucas.py (hunk 5 above)
Comment (lucas-spangher-1) [reply datamodules/lucas.py]: correct, and done

File: datamodules/lucas_processing.py (hunk 7 above)
Comment (lucas-spangher-1) [reply datamodules/lucas_processing.py]: no, I didn't

**Comment by lucas-spangher-1** [comment]
Hey Will sorry, I wound up working on the same branch that I had previously submitted a PR for without realizing. 

This time, I've updated it so that the code takes in train test val inds from csvs (following your guidelines) and also so that the model outputs eval to a common set of codes. 

I think everything is correct.... 

Would be great to have your eyes on it and maybe have you run the sweep, or give me some guidance in how to do so. Some parameters that need to be swept over are: 

- cfg.test.eval_high_thresh
- cfg.test.eval_low_thresh
- cfg.test.eval_hysteresis

File: datamodules/lucas_processing.py (hunk 8 above)
Comment (lucas-spangher-1) [reply datamodules/lucas_processing.py]: Done!
